ext/opentelemetry-ext-docker-tests/tests/pymongo/test_pymongo_functional.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2020, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import typing
+import unittest
+
+from pymongo import MongoClient
+
+from opentelemetry import trace as trace_api
+from opentelemetry.ext.pymongo import trace_integration
+from opentelemetry.sdk.trace import Span, Tracer, TracerSource
+from opentelemetry.sdk.trace.export import SimpleExportSpanProcessor
+from opentelemetry.sdk.trace.export.in_memory_span_exporter import (
+    InMemorySpanExporter,
+)
+
+MONGODB_HOST = os.getenv("MONGODB_HOST ", "localhost")
+MONGODB_PORT = int(os.getenv("MONGODB_PORT ", "27017"))
+MONGODB_DB_NAME = os.getenv("MONGODB_DB_NAME ", "opentelemetry-tests")
+MONGODB_COLLECTION_NAME = "test"
+
+
+class TestFunctionalPymongo(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls._tracer_source = TracerSource()
+        cls._tracer = Tracer(cls._tracer_source, None)
+        cls._span_exporter = InMemorySpanExporter()
+        cls._span_processor = SimpleExportSpanProcessor(cls._span_exporter)
+        cls._tracer_source.add_span_processor(cls._span_processor)
+        trace_integration(cls._tracer)
+        client = MongoClient(
+            MONGODB_HOST, MONGODB_PORT, serverSelectionTimeoutMS=2000
+        )
+        db = client[MONGODB_DB_NAME]
+        cls._collection = db[MONGODB_COLLECTION_NAME]
+
+    def setUp(self):
+        self._span_exporter.clear()
+
+    def validate_spans(self, spans: typing.Tuple[Span]):
+        self.assertEqual(len(spans), 2)
+        for span in spans:
+            if span.name == "rootSpan":
+                root_span = span
+            else:
+                pymongo_span = span
+            self.assertIsInstance(span.start_time, int)
+            self.assertIsInstance(span.end_time, int)
+        self.assertIsNotNone(pymongo_span.parent)
+        self.assertEqual(pymongo_span.parent.name, root_span.name)
```

Comment:
By this point you require `pymongo_span` and `root_span` to be defined. They may not be if `spans` only includes 2 spans whose `name` attribute is `"rootSpan"` or if it includes 2 spans whose name is not. If any of those happen, you'll get a `NameError` in a test case when you want to get only `AssertionError`s. Instead for looping through `spans` looking for spans whose `name` attribute matches, it should be asserted that `spans` actually contains one span whose `name` attribute is `"rootSpan"` and another one whose `name` attribute is not.

Reply:
The actual test I wanted to include is that rootSpan is parent of other spans created, that is why I'm saving the reference for it so I can assert later, the check of an extra span being created in pymongo is already taken care in self.assertEqual(len(spans), 2) because all tests start with a rootSpan only, are you suggesting to add more check regarding these spans?

Reply:
Yes, what I am pointing out is that your test case requires `self._span_exporter.get_finished_spans()` to deliver you 2 spans, one named `"rootSpan"` and other not, and since it requires that, it is probably something that you want to specifically test.